pkg/provider/gitlab.go

```diff
@@ -30,10 +33,25 @@ func (repo *GitLabRepository) Init(config map[string]string) error {
 		gitlabBaseURL = os.Getenv("CI_SERVER_URL")
 	}
 
+	repo.useJobToken = false
 	token := config["token"]
 	if token == "" {
 		token = os.Getenv("GITLAB_TOKEN")
 	}
+	if token == "" {
+		token = os.Getenv("CI_JOB_TOKEN")
+		repo.useJobToken = true
+
+		repo.localRepo = &GitProvider.Repository{}
+		err := repo.localRepo.Init(map[string]string{
```

Comment:
We should confirm that this isn't sparse checkout using 

https://docs.gitlab.com/ee/ci/runners/configure_runners.html#git-strategy

Reply:
So check if `GIT_STRATEGY == none` ? Afaik fetching would also grab the worktree but I'm not sure I got this right

Reply:
that makes sense., because we could conditionally run the fetch. But probably better to just fail anyway instead of doing a sparse checkout and then a fetch immediately afterwards